src/main/java/com/breskul/bring/AnnotationConfigApplicationContext.java

```diff
@@ -0,0 +1,110 @@
+package com.breskul.bring;
+
+import com.breskul.bring.exceptions.BeanInitializingException;
+import com.breskul.bring.exceptions.NoSuchBeanException;
+import com.breskul.bring.exceptions.NoUniqueBeanException;
+import com.breskul.bring.exceptions.NuSuchBeanConstructor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Arrays;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+/**
+ * Standalone application context, accepting component classes as input — in particular @Configuration-annotated classes,
+ * but also plain @Component types
+ */
+public class AnnotationConfigApplicationContext implements ApplicationContext {
+    private static final Logger LOGGER = LoggerFactory.getLogger(BringApplication.class);
+    private final Map<String, Object> context;
+    private final String packageName;
+
+    public AnnotationConfigApplicationContext(String packageName) {
+        Objects.requireNonNull(packageName);
+        this.context = new ConcurrentHashMap<>();
```

Comment:
We can initialize context map not using contructor: 
`private final Map<String, Object> context = new ConcurrentHashMap<>();`